src/features/chat/Chat.tsx

```diff
@@ -0,0 +1,263 @@
+import React, { useEffect, useState, useRef, KeyboardEvent } from 'react'
+// import { useHistory } from 'react-router'
+import cn from 'classnames'
+import routes from '../../common/constants/routes.json'
+import { useHistory } from 'react-router-dom'
+import { ChatItem, ChatItemProps } from './ChatItem'
+import {
+  ChatMessage,
+  ChatMessageProps,
+  DIR_RECV,
+  DIR_SENT,
+} from './ChatMessage'
+import editIcon from '../../common/assets/icons/ico-edit.svg'
+import chatMenuIcon from '../../common/assets/icons/ico-chat-menu.svg'
+import msgEmojiIcon from '../../common/assets/icons/ico-chatmsg-emoji.svg'
+import msgAttachIcon from '../../common/assets/icons/ico-chatmsg-link.svg'
+import msgSendIcon from '../../common/assets/icons/ico-chatmsg-send.svg'
+import chatCloseIcon from '../../common/assets/icons/ico-close2.svg'
+import styles from './Chat.module.css'
+import { mockChats, curUser } from './mock-data'
+
+export type LocationState = {
+  param: string
+}
+
+export default function Chat(): JSX.Element {
+  const [newMsgPlaceholder, setNewMsgPlaceholder] = useState(true)
+  const [newMsg, setNewMsg] = useState('')
+  const [activeChatId, setActiveChatId] = useState<number>(mockChats[0].id)
+  const [activeChat, setActiveChat] = useState<ChatItemProps>(mockChats[0])
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [chats, setChats] = useState<ChatItemProps[]>(mockChats) // DEBUG: init state
+
+  /*
+  const location = useLocation<LocationState>()
+
+  useEffect(() => {
+    if (location.state) {
+      setParam(location.state?.param)
+    }
+  }, [location])
+  */
+
+  const endRef = useRef<HTMLDivElement>(null)
+  const scroll2End = () => {
+    endRef.current?.scrollIntoView({ behavior: 'smooth' })
+  }
+
+  const onNewMsgFocus = () => {
+    setNewMsgPlaceholder(false)
+  }
+
+  const onNewMsgBlur = () => {
+    setNewMsgPlaceholder(newMsg.length > 0 ? false : true)
+  }
+
+  const onNewMsgChange = (ev: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setNewMsg(ev.currentTarget.value)
+  }
+
+  /**
+   * Click by active chat menu button "..."
+   */
```

Comment:
Please remove such comments, function name is supposed to tell what function is doing.

`onActiveChatMenu` - this name doesn't really tell what is this for, how about `toggleChatMenu`?

`onChatEdit`, `onSelectChat` sounds clear to me, no need for comment.

Reply:
comments removed, func renamed